core/src/main/java/nl/nn/adapterframework/webcontrol/api/TransactionalStorage.java

```diff
@@ -800,14 +800,8 @@ public void setCommentMask(String commentMask) {
 
 		public boolean matchMessage(IMessageBrowsingIteratorItem iterItem) throws ListenerException, IOException {
 			if(message != null) {
-				Object rawmsg = storage.browseMessage(iterItem.getId());
-				String msg = null;
-				if (listener != null) {
-					msg = listener.extractMessage(rawmsg, new HashMap<String, Object>()).asString();
-				} else {
-					msg = Message.asString(rawmsg);
-				}
-				if (msg == null || msg.indexOf(message)<0) {
+				String msg = getRawMessage(storage, listener, iterItem.getId());
+				if (msg == null || !StringUtils.containsIgnoreCase(msg, message)) {
```

Comment:
Good find! The only thing I'm a bit worried about is that it does a `Misc.cleanseMessage`. I'm not sure if you should be able to search for 'hidden' values in the message? For now at least, this is a good start :) !

Reply:
Is it possible to create a new class for the filter and add some tests for it?